app/models/event.rb

```diff
@@ -19,4 +19,53 @@ class Event < ApplicationRecord
       )
     end
   end
+
+  def timeslots_with_ranking
+    timeslots_with_score = add_score_to_timeslot(timeslots)
+
+    sorted_timeslots = timeslots_with_score.sort do |a, b|
+      b.score <=> a.score
+    end
+
+    add_rank_to_timeslots sorted_timeslots
+  end
+
+  def add_score_to_timeslot(timeslots)
```

Comment:
Defining singleton methods in-line and having multiple implementations looks pretty spooky to me 🎃.  I need to spend more time reviewing this to understand what is going on.

Reply:
@jeremyjung I refactored by adding a `rank` method to the `Timeslot` model that always returns nil. Now, in the `Event` model, we only override the `rank` method for the 3 best timeslots. Not sure how else to add ranking info to a timeslot instance.